src/firefly/java/edu/caltech/ipac/firefly/server/util/DownloadScript.java

```diff
@@ -91,6 +93,105 @@ public class DownloadScript {
             [ -n "$fileName" ] && opts="wget -O $fileName"
           """);
 
+    static final String zipFunc = """
+          # Unzip logic
+          if [ -n "$fileName" ] && [ -f "$fileName" ] && echo "$fileName" | grep -qE '\\.zip$'; then
+            if command -v unzip &> /dev/null; then
+              unzip -qq "$fileName"
+            fi
+          fi
+        """;
+//====================================================================*/
+
+
+//todo: uncomment this after old script (above) works reliably with IRSA services again, then try the new script below
+//====================================================================
+//  Consistent behavior for both curl and wget:
+//  - Both curl and wget must follow redirects.
+//  - Command exits with non-zero status on failure or error http status code.
+//  - If UrlInfo.filePath is provided:
+//      * If it ends with '/', treat it as a directory.
+//      * Otherwise, treat it as a file path (including possible subdirectories).
+//  - When downloading:
+//      * Save the file to the specified path.
+//      * If no file name is provided, use the remote file name from the URL, or from the Content-Disposition header if available.
+//        * curl will overwrite an existing file, whereas wget will create a new file with a numbered suffix to avoid overwriting.
+//====================================================================
+static final String funcTmpl = """
```

Comment:
This is a little picky so take if or leave it.

I don't think you need to do string substitution 
Try this: at the top of the function 
```bash
curlCmd="curl -fLJO"
wgetCmd="wget --content-disposition"
if [ "$URL_GETTER" = "curl" ]; then
   cmd=$curCmd
else
   cmd=$wgetCmd
fi
```
then right before the downoad_file function set which way. That will have to be in the java code
```java
writer.println( is(Wget, attribs) ? "URL_GETTER=wget" : "URL_GETTER=curl");
write.printlin("");
writer.println(funcTmpl);
```
No string substitution and it allows the use the change the command with out have to go through the while packaging process again.

Reply:
I'll tinker with this in the very next ticket I have (where I change the UI of the download dialog) since that's a smaller ticket anyway if that's ok. Since this PR has been open for some time, I would like to close it to have a working version of the download script across our apps.